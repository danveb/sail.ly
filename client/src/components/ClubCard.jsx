import { Link } from "react-router-dom"; 
import yachtClub from "../assets/clubs/yacht-club.jpg"; 
import "../styles/ClubCard.css"; 

export const ClubCard = ({ name, address, city, state, zip, lat, lon, tel, url, snake }) => {
    return (
        <div className="clubCard" data-testid="clubCard">
            <Link to={`/clubs/${snake}`}>
                <img
                    src={yachtClub}
                    alt="yacht club"
                    className="clubCard__img"
                />
            </Link>
            <div className="clubCard__details">
                <h4>{name}</h4>
                <p>{address}</p>
                <p>{city}, {state} {zip}</p>
                <span>Latitude: {lat.toFixed(2)}°</span>
                <span>Longitude: {lon.toFixed(2)}°</span>
            </div>
        </div>
    )
}