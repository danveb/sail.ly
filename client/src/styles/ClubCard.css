.clubCard {
    display: flex; 
    flex-direction: column; 
    padding: 1rem; 
    gap: 1.5rem; 
}

.clubCard__img {
    width: 100%; 
    height: 100%; 
    object-fit: cover; 
}

.clubCard__details {
    display: flex; 
    flex-direction: column; 
    gap: 0.4rem; 
    color: #000; 
}

.clubCard__details h4 {
    font-family: "Prata", sans-serif; 
    font-size: 22px; 
}

.clubCard__details p {
    font-family: "Yrsa", sans-serif; 
    font-size: 16px; 
}

.clubCard__details span {
    font-weight: 400; 
    font-size: 12px; 
}