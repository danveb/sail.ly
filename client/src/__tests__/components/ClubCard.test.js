import { render, screen } from "@testing-library/react";
import ClubCard from "../../components/ClubCard"; 
import { BrowserRouter as Router } from "react-router-dom"; 

describe("ClubCard Component", () => {
    // smoke test
    test("renders without crashing", () => {
        render(
            <Router>
                <ClubCard />
            </Router>
        );
    });

    // snapshot test
    test("matches snapshot", () => {
        const { asFragment } = render(
            <Router>
                <ClubCard />
            </Router>
        ); 
        expect(asFragment()).toMatchSnapshot(); 
    });

    // getByText
    test("renders Latitude text", () => {
        render(
            <Router>
                <ClubCard />
            </Router>
        );
        const textElement = screen.getByText("Latitude:", { exact: false }); 
        expect(textElement).toBeInTheDocument(); 
    });

    // getByTestId
    test("it renders ClubCard div", () => {
        render(
            <Router>
                <ClubCard />
            </Router>
        );
        const testId = screen.getByTestId("clubCard"); 
        expect(testId).toBeInTheDocument(); 
    });

    // mocking fetch API but test not working... 
    // test.skip("renders clubList data [] returned by fetch API", async () => {
    //     const mockData = [
    //         {
    //             id: 1, 
    //             name: "Soverel Harbour Marina", 
    //             city: "Palm Beach Gardens", 
    //             state: "FL",
    //             zip: 33410,
    //             lat: 26.8460138009802,
    //             lon: -80.0703925850772,
    //             tel: "561-691-9554",
    //             url: "https://soverelmarina.com",
    //             snake: "soverel-harbour-marina",
    //         },
    //         {
    //             id: 2, 
    //             name: "Sunrise Harbor Marina", 
    //             city: "Fort Lauderdale", 
    //             state: "FL",
    //             zip: 33304,
    //             lat: 26.1391917026305,
    //             lon: -80.1093938879056,
    //             tel: "954-667-6720",
    //             url: "https://www.sunriseharbormarina.net",
    //             snake: "sunrise-harbor-marina",
    //         },
    //     ]; 
    //     // global.fetch = jest.fn().mockResolvedValue({
    //     //     json: () => Promise.resolve(mockData), 
    //     // });
    //     jest.spyOn(global, 'fetch').mockResolvedValue({
    //         json: jest.fn().mockResolvedValue(mockData),
    //     });
    //     render(
    //         <Router>
    //             <ClubCard />
    //         </Router>
    //     ); 
    // });
}); 