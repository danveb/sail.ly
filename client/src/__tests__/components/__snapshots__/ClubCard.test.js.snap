// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ClubCard Component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="clubCard"
    data-testid="clubCard"
  >
    <a
      href="/clubs/undefined"
    >
      <img
        alt="yacht club"
        class="clubCard__img"
        src="yacht-club.webp"
      />
    </a>
    <div
      class="clubCard__details"
    >
      <h4>
        <a
          href="/clubs/undefined"
        />
      </h4>
      <p />
      <p>
        ,  
      </p>
      <span>
        Latitude: °
      </span>
      <span>
        Longitude: °
      </span>
    </div>
  </div>
</DocumentFragment>
`;
